nbformat/notebooknode.py

```diff
@@ -1,23 +1,50 @@
 """NotebookNode - adding attribute access to dicts"""
 
 from ipython_genutils.ipstruct import Struct
+from collections import Mapping
+
 
 class NotebookNode(Struct):
     """A dict-like node with attribute-access"""
-    pass
+
+    def __setitem__(self, key, value):
+        if isinstance(value, Mapping):
+            value = from_dict(value)
```

Comment:
This should also check `and not isinstance(value, NotebookNode)` because nodes are dicts, too.